src/frontend/src/scheduler/plan_fragmenter.rs

```diff
@@ -394,11 +394,7 @@ impl BatchPlanFragmenter {
                 if let Some(scan_node) = node.as_batch_seq_scan() {
                     let table_desc = scan_node.logical().table_desc();
 
-                    // FIXME: workaround because batch parallel scan on ordered mv may have unorderd
-                    // results. should do a merge sort after parallel scan.
-                    // Tracking issue: <https://github.com/singularity-data/risingwave/issues/3583>
-                    let is_singleton = true;
-                    // let is_singleton = builder.parallelism == 1;
+                    let is_singleton = builder.parallelism == 1;
```

Comment:
Could you please also make the batch iterator unordered?

https://github.com/singularity-data/risingwave/blob/030a2f80ce77cb1279a3798c20d6bd156b50bcf9/src/storage/src/table/storage_table.rs#L554-L556